### PR TITLE
Change inner product and norm in psd

### PR DESCRIPTION
**Pull request by semlen**
fix issue #128

### CODE REVIEW

**Comment by nkoep** [comment]
The test should be fixed by changing the last line of the test case to
```python
        np.testing.assert_almost_equal(
            np.tensordot(a, b.transpose((0, 2, 1)), axes=a.ndim),
            man.inner(x, multiprod(x, a), multiprod(x, b))
        )
```
as per your PR. This kind of test is always a bit iffy to be honest since it simply reimplements the method in the test, but it'll have to do for now.

File: pymanopt/manifolds/psd.py [hunk 1]
```diff
@@ -76,12 +77,8 @@ def ehess2rhess(self, x, egrad, ehess, u):
...
 
     def norm(self, x, u):
-        # This implementation is as fast as np.linalg.solve_triangular and is
-        # more stable, as the above solver tends to output non positive
-        # definite results.
-        c = la.cholesky(x)
-        c_inv = la.inv(c)
-        return la.norm(multiprod(multiprod(c_inv, u), multitransp(c_inv)))
+        xinvu = la.solve(x, u)
+        return np.sqrt(np.tensordot(xinvu, multitransp(xinvu), axes=x.ndim))
```
Comment (nkoep) [inline pymanopt/manifolds/psd.py:81]: This effectively reimplements `inner` (as is often the case). I'd therefore suggest to change inner to something like this
```python
    def inner(self, x, u, v):
        xinvu = la.solve(x, u)
        if u is v:
            xinvv = xinvu
        else:
            xinvv = la.solve(x, v)
        return np.tensordot(xinvu, multitransp(xinvv), axes=x.ndim)
```
and change `norm` to
```suggestion
        return np.sqrt(self.inner(x, u, v))
```

**Review by semlen — COMMENTED** [review]
Looks good

**Comment by nkoep** [comment]
Some small linting errors left. Also, as by my last comment, I'd prefer to avoid computing `la.solve(x, v)` in `inner` if we can check that `u` and `v` are identical anyway.

**Pull request merged**